presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataType.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.datatype;
+
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.VarcharType;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.facebook.presto.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+public class DataType<T>
+{
+    private Function<T, String> toLiteral;
+    private Function<T, ?> toResult;
```

Comment:
`toResult` should produce an object equal to the object returned by the `select *` query executed on presto. How about `toQueryResult`?